packages/shopping/src/services/refreshtoken.service.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
```

Comment:
@derdeka , cool stuff. I was going to just increase the token lifetime from 10 minutes to 6 hours in my last PR ;).

I am wondering : 
- are you going to add mocha tests?
- have you interacted with the shopping cart API Explorer to see how this will affect the user experience? 
   Developers may want to use the API Explorer to interact with the REST API to debug how things work
   for extended periods of time. It would annoy me when the jwt token would expire in 10 minutes. So I would
   set it for 6 hours while I debug and play with the REST API. How will these changes affect this?
- What needs to change in the Shoppy application (UI front-end) that was recently added?

Reply:
@emonddr All good questions.  Above @derdeka said he'd add tests after initial feedback.

I think the general idea is for the refresh token to hold the session open on a device for a period of time(say 6 hours, two days or even longer) without the need to log back in.  When your access token runs out after 10 minutes, then the refresh token would be used to get new access token without the need to log in again.  This would also need to be designed so that the refresh token can be revoked and force a login. Is that correct @derdeka?

Reply:
> * are you going to add mocha tests?

I'll add some tests.

> * have you interacted with the shopping cart API Explorer to see how this will affect the user experience?

On a `angular` application i would use a `serviceworker` or `service` to periodically refresh the accesstoken. Is there something similar in the shopping example?

>   Developers may want to use the API Explorer to interact with the REST API to debug how things work
>   for extended periods of time. It would annoy me when the jwt token would expire in 10 minutes. So I would
>   set it for 6 hours while I debug and play with the REST API. How will these changes affect this?

Good catch, i have currently no concept for this. My use case is a regular user on a `angular `SPA.

> * What needs to change in the Shoppy application (UI front-end) that was recently added?

I did not have a deep look how the frontend is currently working. I guess a `serviceworker` could work here too?